examples/pytorch_imagenet_resnet50.py

```diff
@@ -156,13 +168,20 @@ def train(epoch):
             if args.cuda:
                 data, target = data.cuda(), target.cuda()
             optimizer.zero_grad()
-            output = model(data)
-            loss = F.cross_entropy(output, target)
-            loss.backward()
+            # Split batch_size * batches_per_allreduce batch into sub-batches
+            # of size batch_size
+            for i in range(0, allreduce_batch_size, args.batch_size):
+                data_batch = data[i:i + args.batch_size]
+                target_batch = target[i:i + args.batch_size]
+                output = model(data_batch)
+                train_accuracy.update(accuracy(output, target_batch))
+                loss = F.cross_entropy(output, target_batch)
+                # Average gradients among sub-batches
```

Comment:
Train loss should be updated before re-scaling loss, since it does its own averaging.  Also, we need to scale by proper number of batches.

```
train_loss.update(loss)
# Average gradients among batches in this data slice.
loss.div_(math.ceil(float(len(data)) / args.batch_size)
```